pkg/diagnostics/v2/client.go

```diff
@@ -108,6 +108,30 @@ func (c *Client) Create() (string, error) {
 	}
 }
 
+// Delete deletes the given cluster bundle.
+func (c *Client) Delete(id string) error {
+	req, err := c.http.NewRequest("DELETE", fmt.Sprintf("%s/%s", baseURL, id), nil)
+	if err != nil {
+		return err
+	}
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
```

Comment:
We can extract this function into a function

Reply:
We're handling the different status codes differently in each method:

* In `Delete` (here): 404 and 304 are handled explicitly and on 200 `nil` is returned.
* In `Create`: On 200 the method decodes the body and returns the ID, all other codes are generic/unknown errors.
* In `Download`: On 200 the body is written to the `dst` parameter, 404 and 500 are handled explicitly.

The extracted function would have to be so generic that IMHO it's not worth it having it in the first place.